### PR TITLE
feat: allow set timeout for gce

### LINKED ISSUE #517: The ComputeEngineCredentials auto-detection timeout is too tight
The autodetection logic in google-auth-library-java uses a 1.5 second timeout (3 tries at 0.5 seconds per try) before deciding that it isn't running on Compute Engine.

Unfortunately, 1.5 seconds is a little tight for GKE Workload Identity / gke-metadata-server, since it needs to wait for IP address information to bounce around between kube-apiserver and the node.

A timeout more like 10 seconds would go a long way towards making the autodetection more stable. Or, perhaps the client library could look for a "I'm definitely running on Compute Engine" environment variable.

### PR DESCRIPTION
**Pull request by arithmetic1728**
Fixes #517 

Allow the GKE users to explicitly set the metadata server timeout via `GCE_METADATA_TIMEOUT` environment variable (this env variable is used by google-auth-library-python as well).

### CODE REVIEW

File: oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java
Comment (chingor13) [inline oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java:90]: Can we prepend `DEFAULT_` to this constant name to reflect that it's now a default.

File: oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java
Comment (chingor13) [inline oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java:80]: Let's confirm that this behavior and environment name will be consistent across all 7 auth libraries

File: oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java
Comment (arithmetic1728) [reply oauth2_http/java/com/google/auth/oauth2/ComputeEngineCredentials.java]: done

File: oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java
Comment (arithmetic1728) [reply oauth2_http/java/com/google/auth/oauth2/DefaultCredentialsProvider.java:80]: This is the env variable name and behavior as python (https://github.com/googleapis/google-auth-library-python/blob/master/google/auth/compute_engine/_metadata.py#L58). The rest auth libs are still using a default value. The behavior is already not consistent for more than 4 years.

This is a cheap fix to unblock GKE users while we figure out if we can do a better fix, there will be some work across 7 auth libs, either this fix or some better fix.

**Review by silvolu — APPROVED** [review]

**Review by chingor13 — CHANGES_REQUESTED** [review]
New environment variables need to be clearly documented across the languages. Can we get this in an AIP? go/std-client-env-vars

**Comment by chingor13** [comment]
Also closing this one like .NET. We will revisit once the env variable is approved across the board.

**Pull request closed without merging**
